### PR TITLE
Add saltifier

### PR DESCRIPTION
**Pull request by Warchant**
Main purpose is to change **each value** in JSON into new JSON:

```
{
  "v": <old value>,
  "s": <salt>
}
```

### CODE REVIEW

File: src/main/java/jp/co/soramitsu/sora/json/Saltifier.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
+package jp.co.soramitsu.sora.json;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+public class Saltifier {
+
+  private final ObjectMapper mapper;
```
[the remark this answers is not in the document]
Comment (Warchant) [reply src/main/java/jp/co/soramitsu/sora/json/Saltifier.java]: Codacy works bad with this annotation, so this is `won't fix`

[the remark this answers is not in the document]
Comment (rogachevsn55) [reply src/main/java/jp/co/soramitsu/sora/json/Saltifier.java]: I still suggest using this annotation and change the Codacy rules

File: src/main/java/jp/co/soramitsu/sora/json/Saltifier.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+@FieldDefaults(makeFinal = true, level = AccessLevel.PRIVATE)
+public class Saltifier {
+
+  ObjectMapper mapper;
+  SaltGenerator generator;
+  String valueKey;
+  String saltKey;
+
+  public Saltifier(ObjectMapper mapper, SaltGenerator generator) {
+    this(mapper, generator, "v", "s");
```
Comment (tdorzhi) [inline src/main/java/jp/co/soramitsu/sora/json/Saltifier.java]: better to have `v` and `s` changed to explicit values `value`, `salt`

File: src/main/java/jp/co/soramitsu/sora/json/Saltifier.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  }
+
+  /**
+   * Add salt to the every *value* field in the input JSON.
```
Comment (tdorzhi) [inline src/main/java/jp/co/soramitsu/sora/json/Saltifier.java]: value field is every field except array?

but from example I see that they're saltified too:
```
+  "empty array" : {
+    "v" : [ ],
+    "s" : "salt"
+  },
```

File: src/main/java/jp/co/soramitsu/sora/json/Saltifier.java (hunk 2 above)
Comment (Warchant) [reply src/main/java/jp/co/soramitsu/sora/json/Saltifier.java]: shorter names reduce storage requirements for mobile devices and sora backend

**Pull request merged**